packages/aztec.js/test/proof/privateRange/index.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable prefer-destructuring */
+const { errors } = require('@aztec/dev-utils');
+const secp256k1 = require('@aztec/secp256k1');
+const { expect } = require('chai');
+const { randomHex } = require('web3-utils');
+
+const { PrivateRangeProof } = require('../../../src/proof');
+const note = require('../../../src/note');
+
+describe('Private range proof', () => {
+    const { publicKey } = secp256k1.generateAccount();
+    const sender = randomHex(20);
+    let originalNote = {};
+    const originalNoteValue = 10;
+    let comparisonNote = {};
+    const comparisonNoteValue = 4;
+    let utilityNote = {};
+    const utilityNoteValue = 6;
+
+    beforeEach(async () => {
+        originalNote = await note.create(publicKey, originalNoteValue);
+        comparisonNote = await note.create(publicKey, comparisonNoteValue);
+        utilityNote = await note.create(publicKey, utilityNoteValue);
+    });
+
+    describe('Success States', () => {
+        it('should construct a private range proof with well-formed outputs', async () => {
+            const proof = new PrivateRangeProof(originalNote, comparisonNote, utilityNote, sender);
+
+            const numNotes = 3;
+            expect(proof.data.length).to.equal(numNotes);
+            expect(proof.challengeHex.length).to.equal(66);
+            expect(proof.data[0].length).to.equal(6);
+            expect(proof.data[1].length).to.equal(6);
+            expect(proof.data[2].length).to.equal(6);
+        });
+    });
+
+    describe('Failure States', () => {
```

Comment:
We should test that the proof fails if the input note is not within the desired range.

This should fail in two locations - when constructing the proof, we should return an error state, and we should also test that verifying such a proof fails

Reply:
nevermind, was looking at the wrong file!

Reply:
Good point. Have implemented this check for the privateRange proof construction, but ideally we should also do this for all proofs